### DISCUSSION TITLE
Event is not returning id on creation.

### DISCUSSION THREAD

**Discussion by shoover56** [spatie/laravel-google-calendar#276] [Q&A]
Creating a event using this block of code:

    $event = new Event;
    $event->name = 'Test Event';
    $event->startDateTime = Carbon::createFromTimestamp(strtotime(date('Y-m-d')));
    $event->endDateTime = Carbon::createFromTimestamp(strtotime(date('Y-m-d', time()+(60*60*24))));
    $event->setColorId(6);
    $event->description = 'Test Event';
    $event->save();
    print_r($event);

Returns this:

    Spatie\GoogleCalendar\Event Object
    (
      [googleEvent] => Google\Service\Calendar\Event Object
        (
            [internal_gapi_mappings:protected] => Array
                (
                )

            [modelData:protected] => Array
                (
                )

            [processed:protected] => Array
                (
                )

            [collection_key:protected] => recurrence
            [anyoneCanAddSelf] => 
            [attachmentsType:protected] => Google\Service\Calendar\EventAttachment
            [attachmentsDataType:protected] => array
            [attendeesType:protected] => Google\Service\Calendar\EventAttendee
            [attendeesDataType:protected] => array
            [attendeesOmitted] => 
            [colorId] => 6
            [conferenceDataType:protected] => Google\Service\Calendar\ConferenceData
            [conferenceDataDataType:protected] => 
            [created] => 
            [creatorType:protected] => Google\Service\Calendar\EventCreator
            [creatorDataType:protected] => 
            [description] => Test Event
            [endType:protected] => Google\Service\Calendar\EventDateTime
            [endDataType:protected] => 
            [endTimeUnspecified] => 
            [etag] => 
            [eventType] => 
            [extendedPropertiesType:protected] => Google\Service\Calendar\EventExtendedProperties
            [extendedPropertiesDataType:protected] => 
            [gadgetType:protected] => Google\Service\Calendar\EventGadget
            [gadgetDataType:protected] => 
            [guestsCanInviteOthers] => 
            [guestsCanModify] => 
            [guestsCanSeeOtherGuests] => 
            [hangoutLink] => 
            [htmlLink] => 
            [iCalUID] => 
            [id] => 
            [kind] => 
            [location] => 
            [locked] => 
            [organizerType:protected] => Google\Service\Calendar\EventOrganizer
            [organizerDataType:protected] => 
            [originalStartTimeType:protected] => Google\Service\Calendar\EventDateTime
            [originalStartTimeDataType:protected] => 
            [privateCopy] => 
            [recurrence] => Array
                (
                    [summary] => Test Event
                    [description] => Test Event
                )

            [recurringEventId] => 
            [remindersType:protected] => Google\Service\Calendar\EventReminders
            [remindersDataType:protected] => 
            [sequence] => 
            [sourceType:protected] => Google\Service\Calendar\EventSource
            [sourceDataType:protected] => 
            [startType:protected] => Google\Service\Calendar\EventDateTime
            [startDataType:protected] => 
            [status] => 
            [summary] => Test Event
            [transparency] => 
            [updated] => 
            [visibility] => 
            [workingLocationPropertiesType:protected] => Google\Service\Calendar\EventWorkingLocationProperties
            [workingLocationPropertiesDataType:protected] => 
            [start] => Google\Service\Calendar\EventDateTime Object
                (
                    [internal_gapi_mappings:protected] => Array
                        (
                        )

                    [modelData:protected] => Array
                        (
                        )

                    [processed:protected] => Array
                        (
                        )

                    [date] => 
                    [dateTime] => 2023-09-13T00:00:00-05:00
                    [timeZone] => America/Chicago
                )

            [end] => Google\Service\Calendar\EventDateTime Object
                (
                    [internal_gapi_mappings:protected] => Array
                        (
                        )

                    [modelData:protected] => Array
                        (
                        )

                    [processed:protected] => Array
                        (
                        )

                    [date] => 
                    [dateTime] => 2023-09-14T00:00:00-05:00
                    [timeZone] => America/Chicago
                )

        )

    [calendarId:protected] => 
    [attendees:protected] => Array
        (
        )

    [hasMeetLink:protected] => 
    )

The ID value not being set on return is important. We need to be able to keep the ids of events for different activities we do. I believe a temporary workaround is after creating the event I reacquire all events and iterate all of them until I got the one that matches the input criteria. Since I can't be sure of any significant limitations if it adds up to hundreds, I figured I could ask if I need to do anything to make it return an id.

**Comment by AIC-BV** [comment]
Hi
I will be needing this too very soon.
Are you still stuck on this?

**Reply by shoover56** [reply]
Yes. I can let you know in this issue if I make any progress or get any workarounds.

**Comment by AIC-BV** [comment]
> [id] =>

Just wanted to let you know that ID is set, but it is a string, not an integer. Works for me.

**Comment by BayronVazquez** [comment]
Hi, I had the same problem and the solution I found was this

```
$event = Event::create([
    'name'            => "TITLE",
    'description'     => "SOME DESCRIPTION",
    'startDateTime'   => Carbon::parse("{$fecha_inicio} {$hora_inicio}"),
    'endDateTime'     => Carbon::parse("{$fecha_final} {$hora_final}"),
], $ID_GOOGLE_CALENDAR);
```
By using this way you force the library to send the HTTP request to the Google API at that moment 
and thus you obtain the Event object with all the corresponding data.

**Reply by shoover56** [reply]
Appreciate it @BayronVazquez ! My backup solution is just after creating the event to get all the upcoming events and just iterate until the name, desc, start/end times match up with what I originally sent.

**Comment by jackleblack-kpt** [comment]
You need to assign the result of event save 👍 : 
`  $event = $event->save();
   logger($event->id);`

**Reply by ievgen-klymenko-uvoteam** [reply]
Thank you. That took me some time to find..
